### PR TITLE
Remove support for deprecated StoredScript contexts

### PR DESCRIPTION
**Pull request by sohaibiftikhar**
-- related to #27612
-- follows up on #28939

### CODE REVIEW

**Comment by imotov** [comment]
add to whitelist

**Review by nik9000 — COMMENTED** [review]
This looks right to me. @rjernst, since you know this code better than I do, can you also take a look?

File: server/src/main/java/org/elasticsearch/script/StoredScriptSource.java [hunk 1]
```diff
@@ -329,28 +307,9 @@ public static StoredScriptSource parse(BytesReference content, XContentType xCon
...
-
-                    if (source == null || source.isEmpty()) {
-                        DEPRECATION_LOGGER.deprecated("empty templates should no longer be used");
-                    }
-
-                    return new StoredScriptSource(Script.DEFAULT_TEMPLATE_LANG, source, Collections.emptyMap());
-                } else {
-                    return parseRemaining(token, parser);
-                }
-            } else if (TEMPLATE_NO_WRAPPER_PARSE_FIELD.getPreferredName().equals(name)) {
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/script/StoredScriptSource.java]: `TEMPLATE_NO_WRAPPER_PARSE_FIELD` and `TEMPLATE_PARSE_FIELD` constants can be removed now right?

**Review by rjernst — COMMENTED** [review]
This looks good. I have one comment.

File: server/src/main/java/org/elasticsearch/script/StoredScriptSource.java (hunk 1 above)
Comment (sohaibiftikhar) [reply server/src/main/java/org/elasticsearch/script/StoredScriptSource.java]: Yes, I already did this. I will commit this together with the fix for the failing test.

**Comment by sohaibiftikhar** [comment]
@nik9000 @rjernst Could you guys have a looks again when you have time?

**Review by nik9000 — COMMENTED** [review]
Looks good to me. Could you add something to `migrate_7_0/api.asciidoc` to document that this deprecated format is now unsupported?

**Pull request merged**